### PR TITLE
Refactor the libcgroups interface

### PR DESCRIPTION
**Pull request by yihuaf**
- Refactored the cgroup creation into a struct instead of individual parameters. This reduce the amount of container_args field we have to carry across the process boundry.
- Refactored the use of &Path vs. PathBuf in the interface.
- Tag along since it is too small. Minor updates to the notify listener comments and variable name. Also makes it clone-able.

I have 3 more PR.

### CODE REVIEW

File: crates/libcgroups/src/common.rs [hunk 1]
```diff
@@ -316,65 +316,70 @@ pub enum CreateCgroupSetupError {
...
-pub fn create_cgroup_manager<P: Into<PathBuf>>(
-    cgroup_path: P,
-    systemd_cgroup: bool,
-    container_name: &str,
+#[derive(Clone)]
+pub struct CgroupConfig {
+    pub cgroup_path: PathBuf,
+    pub systemd_cgroup: bool,
+    pub container_name: String,
+}
```
Comment (utam0k) [inline crates/libcgroups/src/common.rs:324]: What's the benefit of this change?

**Comment by utam0k** [comment]
This is a breaking change to libcgroup, not libcontainer, right?

File: crates/libcgroups/src/common.rs (hunk 1 above)
Comment (yihuaf) [reply crates/libcgroups/src/common.rs:324]: This reduces the number of variables I have to store inside container_args. This PR moves the creation of the cgroup manager to the first thing in intermediate process, so we have to store the variables needed to call cgroup create manager. I can either store 3 or store a 1 struct. Also, these variables are easily shown to be clone-able.

File: crates/libcontainer/src/notify_socket.rs [hunk 2]
```diff
@@ -93,6 +96,15 @@ impl NotifyListener {
...
 }
 
+impl Clone for NotifyListener {
+    fn clone(&self) -> Self {
+        let fd = self.socket.as_raw_fd();
+        // This is safe because we just duplicate a valid fd.
+        let socket = unsafe { UnixListener::from_raw_fd(fd) };
+        Self { socket }
+    }
+}
```
Comment (utam0k) [inline crates/libcontainer/src/notify_socket.rs:106]: 👍 but may I ask you to add the test to ensure the same fs?

**Comment by yihuaf** [comment]
> This is a breaking change to libcgroup, not libcontainer, right?

Yes.

File: crates/libcontainer/src/notify_socket.rs (hunk 2 above)
Comment (yihuaf) [reply crates/libcontainer/src/notify_socket.rs]: Added a unit test to make sure the cloned listener still functions correctly.

File: crates/libcgroups/src/common.rs (hunk 1 above)
Comment (utam0k) [reply crates/libcgroups/src/common.rs:324]: I didn't say enough. I think making it a structure is a good change. However, I doubt that the type is changing🤔

Comment (yihuaf) [reply crates/libcgroups/src/common.rs:324]: > I doubt that the type is changing

I am not sure I follow what you are saying here? Can you rephrase/elaborate?
